### PR TITLE
add cache extensions

### PR DESCRIPTION
**Pull request by jennyf19**
@jmprieur  not sure if you want to leave all the examples of distributed cache, but it looks cleaner when doing `git clone`.

### CODE REVIEW

File: ConfidentialClientTokenCache/Program.cs [hunk 1]
```diff
@@ -64,8 +49,46 @@ static async Task Main(string[] args)
                 .WithTenantId(tenant)
                 .Build();
 
-            msalTokenCacheProvider.Initialize(app.UserTokenCache);
-            msalTokenCacheProvider.Initialize(app.AppTokenCache);
+            // Distributed in memory token caches (App and User caches)
+            app.AddInMemoryTokenCaches();
```
Comment (jmprieur) [inline ConfidentialClientTokenCache/Program.cs]: maybe we'd want to use a #define 
Something like:
#define SERIALIZATION_IMPLEMENTATION InMemory // Or DistributedInMemory, Redis, Sql, CosmosDb

then:

```CSharp
#if SERIALIZATION_IMPLEMENTATION InMemory
            // Distributed in memory token caches (App and User caches)
            app.AddInMemoryTokenCaches();
#endif 

#if DistributedInMemory
             // Distributed token caches (App and User caches)
            // Add one of the below: SQL, Redis, CosmosDb
            app.AddDistributedTokenCaches(services =>
            {
             services.AddDistributedInMemoryCache();
            });
#endif 

```

etc ...

**Review by jmprieur — APPROVED** [review]
LGTM.
I think we'd want the sample to work out of the box, though

File: ConfidentialClientTokenCache/Program.cs (hunk 1 above)
Comment (jennyf19) [reply ConfidentialClientTokenCache/Program.cs]: some people don't know what #define means so i'll just comment it.

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks @jennyf19

**Pull request merged**